charger/heating/powermode.go

```diff
@@ -0,0 +1,54 @@
+package heating
+
+import (
+	"context"
+
+	"github.com/evcc-io/evcc/api"
+)
+
+// PowerModeController controller implementation
+type PowerModeController struct {
+	ctrl      *PowerController
+	maxPowerG func() (int64, error)
+}
+
+// NewPowerModeController creates power mode controller
+func NewPowerModeController(ctx context.Context, ctrl *PowerController, maxPowerG func() (int64, error)) *PowerModeController {
+	return &PowerModeController{
+		ctrl:      ctrl,
+		maxPowerG: maxPowerG,
+	}
+}
+
+func (wb *PowerModeController) power() (int64, error) {
+	if wb.maxPowerG == nil {
+		return wb.ctrl.power, nil
+	}
+	return wb.maxPowerG()
+}
+
+// Status implements the api.Charger interface
+func (wb *PowerModeController) Status() (api.ChargeStatus, error) {
+	power, err := wb.power()
+	if err != nil {
+		return api.StatusNone, err
+	}
+
+	status := map[bool]api.ChargeStatus{false: api.StatusB, true: api.StatusC}
```

Comment:
So richtig passt der ChargeStatus für mich hier nicht.
Eine WP ist ja z. B. quasi nie "aus" oder "an". "Aus" z. B. oft nur wenn man ihr wirklich den Strom abdreht. Sonst läuft in der Regel mindestens mal die UWP etc.
Andererseits gibt es auch oft gar keine Möglichkeit lesend herauszufinden ob das System selbst "normal" vor sich hinarbeitet oder "verstärkt" heizt. Sie macht einfach autark ihr Ding nach Bedarf und alle externen Einflüsse sind nur Empfehlungen.

Finden wir da noch ein besseres Modell?

Reply:
Für mich auch nicht, aber das _ist_ heute beim SGReady Charger schon so.

Reply:
Perspektivische Idee: 
Immer "an". Status B oder C in Abhängigkeit von Power. 
Wenn "boost" eingeschaltet, dann entsprechende Anzeige; analog "climater active" beim Auto?

Reply:
Alles prima, aber hier gehts nicht um den SG Ready charger.